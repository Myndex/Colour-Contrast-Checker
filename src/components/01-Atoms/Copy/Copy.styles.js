import styled, { css } from 'styled-components';

const Copy = styled.span`
  display: inline-block;
  max-width: 575px;

  ${props => props.normal && css`
    font-size: 1em;
  `}

  ${props => props.large && css`
    font-size: 14pt;
  `}
`;

export default Copy;
