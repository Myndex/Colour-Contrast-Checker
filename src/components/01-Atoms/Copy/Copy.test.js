import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import Copy from './Copy.styles';

it('Should render text with normal font size prop', () => {
  const component = <Copy normal>Test Copy</Copy>;
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', '1em');
});

it('Should render text with large font size prop', () => {
  const component = <Copy large>Test Copy</Copy>;
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', '14pt');
});