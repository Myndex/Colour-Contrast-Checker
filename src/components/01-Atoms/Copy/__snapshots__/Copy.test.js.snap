// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render text with large font size prop 1`] = `
.c0 {
  display: block;
  width: 100%;
  max-width: 575px;
  height: auto;
  margin-top: 18px;
  padding-top: 0;
  padding-right: 0;
  padding-bottom: 6px;
  padding-left: 0;
  border: none;
  background-color: transparent;
  line-height: inherit;
  resize: none;
  font-size: 14pt;
}

<textarea
  className="c0"
>
  Test Copy
</textarea>
`;

exports[`Should render text with normal font size prop 1`] = `
.c0 {
  display: block;
  width: 100%;
  max-width: 575px;
  height: auto;
  margin-top: 18px;
  padding-top: 0;
  padding-right: 0;
  padding-bottom: 6px;
  padding-left: 0;
  border: none;
  background-color: transparent;
  line-height: inherit;
  resize: none;
  font-size: 1em;
}

<textarea
  className="c0"
>
  Test Copy
</textarea>
`;
